### PR TITLE
Replace i18n editing logic to not use addMessages

### PR DESCRIPTION
**Pull request by alcar**
#### What problem is this solving?

When editing i18n fields, there are a few scenarios in which the i18n key itself is being saved/displayed; this PR addresses that issue.

#### How should this be manually tested?

[Workspace](https://fixi18nsaving--alssports.myvtex.com/admin/app/cms/site-editor/)

#### Screenshots or example usage

Edit any block with i18n fields, e.g. "Rich Text".

#### Type of changes

| ✔️  | Type of Change                                                                            |
| --- | ----------------------------------------------------------------------------------------- |
| ✔️  | Bug fix                                |
| \_  | New feature                        |
| \_  | Breaking change  |
| ✔️  | Technical improvements  |

#### Notes

While there isn't currently an efficient way to update the inner Runtime's intl, this solution uses the `IOMessage`'s fallback behavior to modify i18n properties. Since the value actually represents an i18n key, typing in the name of an existing one would cause the block to render the corresponding intl string. In order to prevent that, `I18nInput` now adds an invisible character after the input value, so it will (probably) never match an i18n key. The invisible character is removed once the form is submitted, before being sent to the server.

### CODE REVIEW

File: react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx [hunk 1]
```diff
@@ -118,7 +118,6 @@ const ComponentEditor: React.FunctionComponent<Props> = ({
         >
           <Form
             formContext={{
-              addMessages: iframeRuntime.addMessages,
               isLayoutMode: editor.mode === 'layout',
               messages: iframeRuntime.messages,
```
Comment (thiagomurakami) [inline react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx]: Do you still need the to pass the `messages` here?

File: react/utils/components/index.ts [hunk 2]
```diff
@@ -283,12 +285,24 @@ const keepTheBlanks = (
...
+  text + INVISIBLE_CHARACTER
+
+const removeInvisibleCharacter = (text: string) => {
+  const lastIndex = text.length - 1
+
+  if (text[lastIndex] !== INVISIBLE_CHARACTER) {
+    return text
+  }
+
+  return text.slice(0, lastIndex)
```
Comment (thiagomurakami) [inline react/utils/components/index.ts]: Are you appending more than one `INVISIBLE_CHARACTER` to the text? If you're appending `INVISIBLE_CHARACTER` on every onChange, you could end up with the string:
`lalal${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}`, couldn't you?

If so, the `text.length - 1` isn't reliable, I'd use `indexOf` to get the first (in line 292). Also, I would change `.slice` to `.replace` here, as it's safer (to cases like multiple appends, and other cases like `a${INVISIBLE_CHARACTER}b${INVISIBLE_CHARACTER}`). What do you think?

Comment (alcar) [reply react/utils/components/index.ts]: There's always at most only one invisible character.

The `newValue` from [this line](https://github.com/vtex-apps/admin-pages/pull/279/files#diff-aebfa0473e42ecce8b1a6dec918ef42fR47) refers to the new value in the input, and it never contains an invisible character – unless the user added it himself, in which case we shouldn't interfere.

File: react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx (hunk 1 above)
Comment (alcar) [reply react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx]: Nope. Thanks!

File: react/utils/components/index.ts (hunk 2 above)
Comment (alcar) [reply react/utils/components/index.ts]: Come to think of it, [`removeInvisibleCharacter`](https://github.com/vtex-apps/admin-pages/pull/279/files#diff-9b3ddc4469aa6462dfe328035ea0c290R291) will always remove the last invisible character, even if it was added by the user. I don't think people will actually put trailing invisible characters in their strings, but you never know; I like to believe we'll have more efficient ways to update the messages before this becomes an issue.

What do you think?

Comment (thiagomurakami) [reply react/utils/components/index.ts]: > I don't think people will actually put trailing invisible characters in their strings, but you never know

I agree and had the feeling that you'd comment this ⛏ . What I think could happen is the user copy+pasting from the iframe and appending multiple invisible characters, ergo, my suggestion to use `.replace`:

Screen Shot 2019-09-17 at 12 03 37

Comment (alcar) [reply react/utils/components/index.ts]: Hmm, the user would have to copy from the iframe while editing a block, but yes, that could happen. Okay, I'll use `replace` instead!

**Pull request merged**